### PR TITLE
Insert empty line before/after the embed only if the embed is the first/last item in the editor

### PR DESCRIPTION
**Pull request by tomaskikutis**
SDESK-2165

### CODE REVIEW

File: scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+/*
+    insertAtomicBlockWithoutEmptyLinesAroundIt: function(
+        editorState: EditorState,
+        entityKey: string,
+        character: string
+    ): EditorState
+*/
+
+function insertAtomicBlockWithoutEmptyLinesAroundIt(editorState, entityKey, character) {
```
Comment (petrjasek) [inline scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js]: as much as I like descriptive names this seems a bit too much :wink:

File: scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    var selectionState = editorState.getSelection();
+    var target = getInsertionTarget(editorState.getCurrentContent(), selectionState);
+    var asAtomicBlock = Modifier.setBlockType(target.contentState, target.selectionState, 'atomic');
+    var charData = CharacterMetadata.create({entity: entityKey});
+    var fragmentArray = [];
+
+    if (asAtomicBlock.getFirstBlock().getKey() === target.selectionState.getAnchorKey()) {
+        fragmentArray.push(new ContentBlock({
```
Comment (petrjasek) [inline scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js]: we keep annotations/comments data on first block atm, would this still work if there is new block added?

File: scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js (hunk 1 above)
Comment (tomaskikutis) [reply scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js]: I wanted to make it clear it's not the same as the original `insertAtomicBlock` from `draft-js` helpers I based this one on. Do you suggest to keep the name the same?

Comment (petrjasek) [reply scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js]: I think `insertAtomicBlockWithoutEmptyLines` would be enough

File: scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js (hunk 2 above)
Comment (tomaskikutis) [reply scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js]: I've tried to break it by inserting a table in various positions when having active highlights. Seems to work fine.
highlights

**Pull request merged**